src/client.test.ts

```diff
@@ -37,7 +40,7 @@ describe('client APIs', () => {
       },
     });
 
-    const client = new APIClient(integrationConfig);
+    const client = new APIClient(integrationConfig, createMockIntegrationLogger());
```

Comment:
Could we get some test coverage around the new code block? 
If it's not easy, probably not worth it.

Reply:
I've kicked around a few ideas, but wasn't terribly happy with anything yet.  Certainly open to further feedback.  I thought about a simple test to make sure it waited the proper number of seconds, but TBH I've never performed that kind of test to make sure it awaited the proper amount of milliseconds and wasn't even sure that would be worth the effort.